src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -859,17 +858,16 @@ int D4NFilterObject::D4NFilterReadOp::D4NFilterGetCB::handle_data(bufferlist& bl
 	    existing_block.blockID = block.blockID;
 	    existing_block.size = block.size;
 
-	    if ((ret = blockDir->get(dpp, &existing_block, *y)) == -ENOENT) {
-              if ((ret = blockDir->set(dpp, &block, *y)) < 0) 
-                ldpp_dout(dpp, 0) << "D4NFilterObject::D4NFilterReadOp::D4NFilterGetCB::" << __func__ << "(): BlockDirectory set() method failed, ret=" << ret << dendl;
-	    } else if (ret == 0) {
-	      if (existing_block.version != block.version) {
-		if ((ret = blockDir->set(dpp, &block, *y)) < 0) //new versioned block will have new version, hostsList etc, how about globalWeight?
-		  ldpp_dout(dpp, 0) << "D4NFilterObject::D4NFilterReadOp::D4NFilterGetCB::" << __func__ << "(): BlockDirectory set() method failed, ret=" << ret << dendl;
-	      } else {
-		if ((ret = blockDir->set(dpp, &block, *y)) < 0)
-		  ldpp_dout(dpp, 0) << "D4NFilterObject::D4NFilterReadOp::D4NFilterGetCB::" << __func__ << "(): BlockDirectory set() method failed, ret=" << ret << dendl;
-	      }
+	    if ((ret = blockDir->get(dpp, &existing_block, *y)) == 0 || ret == -ENOENT) {
+              if (ret == 0) { //new versioned block will have new version, hostsList etc, how about globalWeight?
+		block = existing_block;
```

Comment:
just check if we want to carry over the dirty flag etc also, but otherwise looks fine